src/test/java/org/commcare/formplayer/tests/MultiSelectCaseClaimTest.java

```diff
@@ -0,0 +1,113 @@
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.util.MultiValueMap;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ *
+ */
+@WebMvcTest
+@ContextConfiguration(classes = TestContext.class)
+public class MultiSelectCaseClaimTest extends BaseTestClass {
+
+    private static final String APP_NAME = "case_claim_with_multi_select";
+    @Autowired
+    CacheManager cacheManager;
+
+    @Captor
+    ArgumentCaptor<String> urlCaptor;
+
+    @Captor
+    ArgumentCaptor<MultiValueMap<String, String>> requestDataCaptor;
+
+    @Override
+    @BeforeEach
+    public void setUp() throws Exception {
+        super.setUp();
+        configureRestoreFactory("caseclaimdomain", "caseclaimusername");
+        cacheManager.getCache("case_search").clear();
+    }
+
+    @Override
+    protected String getMockRestoreFileName() {
+        return "restores/caseclaim.xml";
+    }
+
+    @Test
+    public void testCaseClaimWithMultiSelectList() throws Exception {
+        configureQueryMock();
+        configureSyncMock();
+
+        // default search is on so we should skip to search results directly
+        EntityListResponse entityResp = sessionNavigateWithQuery(new String[]{"1", "action 1"},
+                APP_NAME,
+                null,
+                EntityListResponse.class);
+        assertTrue(entityResp.isMultiSelect());
+
+        String[] selectedValues =
+                new String[]{"94f8d030-c6f9-49e0-bc3f-5e0cdbf10c18", "0156fa3e-093e-4136-b95c-01b13dae66c7",
+                        "0156fa3e-093e-4136-b95c-01b13dae66c8"};
+        String[] selections = new String[]{"1", "action 1", MultiSelectEntityScreen.USE_SELECTED_VALUES};
+        CommandListResponseBean commandResponse = sessionNavigateWithQuery(selections,
+                APP_NAME,
+                null,
+                selectedValues,
+                CommandListResponseBean.class);
+
+        // `use_selected_values' should be replaced in returned selections
+        Assertions.assertNotEquals(selections, commandResponse.getSelections());
+
+        // Verify case claim request
+        verify(webClientMock, times(1)).post(urlCaptor.capture(), requestDataCaptor.capture());
+        assertEquals("http://localhost:8000/a/test/phone/claim-case/", urlCaptor.getAllValues().get(0));
+        MultiValueMap<String, String> requestData = requestDataCaptor.getAllValues().get(0);
+
+        // cases that are owned should not be in the claim request
+        List<String> casesToBeClaimed = Arrays.asList("0156fa3e-093e-4136-b95c-01b13dae66c7",
+                "0156fa3e-093e-4136-b95c-01b13dae66c8");
+        assertEquals(requestData.get("case_id"), casesToBeClaimed);
+    }
+
+    private void configureSyncMock() {
+        when(webClientMock.post(anyString(), any()))
+                .thenReturn("");
+    }
+
+    private void configureQueryMock() {
+        when(webClientMock.postFormData(anyString(), any(Multimap.class)))
```

Comment:
I also notice that this is mocking the `postFormData` call but I think the claim request just uses `post`.

Looking more at the test I think you've got the naming of the mock methods swapped. This one should be `configureSyncMock`.

Reply:
`configureQueryMock` is supposed to mock the case search request which is of type `postFormData`. `configureSyncMock` is for case claim which is as you said `post`.

Reply:
I see, so the names are derived from the names of the 'screen' classes. I don't find the names line up very well with what they do so I'd recommend a rename of these functions:

* `configureSyncMock` -> `configureCaseClaimMockRequest`
* `configureQueryMock` -> `configureCaseSearchMockRequest`

Reply:
+1 for renaming "sync" to either "claim" or "post"